tests/ci/check_so_interface.py

```diff
@@ -0,0 +1,352 @@
+import ast
+# functions and their arguments to check
+# specified here: https://www.overleaf.com/project/5e837cac9659910001e5f71e
+interface = {
+    "python/pysmurf/client/tune/smurf_tune.py": {  # file
+        "SmurfTuneMixin": {                        # class
+            "methods": [
```

Comment:
what is this `methods` item used for? We probably only want to check for the existence of the methods that are listed in the interface document, and then this list is redundant with the individual method specs.